### PR TITLE
Implemented S3 RepositoryProtectedEntityManager

### PR DESCRIPTION
**Pull request by dsu-igeek**
Handles multiple PE types in an S3 bucket

Signed-off-by: Dave Smith-Uchida <dsmithuchida@vmware.com>

### CODE REVIEW

File: pkg/s3repository/repository_protected_entity_manager.go [hunk 1]
```diff
@@ -16,6 +16,94 @@
...
+		prefix = prefix + "/"
+	}
+	loi := s3.ListObjectsV2Input{
+		Bucket:    &bucket,
+		Delimiter: &delimiter,
+		Prefix:    &prefix,
+	}
+	loo, err := s3Client.ListObjectsV2(&loi)
+	if err != nil {
+		return nil, errors.WithMessage(err, "Failed retrieving prefixes")
```
Comment (deepakkinni) [inline pkg/s3repository/repository_protected_entity_manager.go:53]: fix error message, maybe failed retrieving objects with loi params

File: pkg/s3repository/repository_protected_entity_manager.go [hunk 2]
```diff
@@ -16,6 +16,94 @@
...
+	}
+	return petm.GetProtectedEntity(ctx, id)
+}
+
+func (recv RepositoryProtectedEntityManager) GetProtectedEntityTypeManager(peType string) astrolabe.ProtectedEntityTypeManager {
+	recv.mutex.Lock()
+	defer recv.mutex.Unlock()
+	returnPETM := recv.petms[peType]
+	if returnPETM == nil {
+		var err error
```
Comment (deepakkinni) [inline pkg/s3repository/repository_protected_entity_manager.go:90]: probably don't need to declare the error var

File: pkg/s3repository/repository_protected_entity_manager.go [hunk 3]
```diff
@@ -16,6 +16,94 @@
...
+		prefix:  prefix,
+		logger:  logger,
+	}
+	for _, curPrefix := range loo.CommonPrefixes {
+		curPrefixStr := curPrefix.String()
+		components := strings.Split(curPrefixStr, "/")
+		curTypeName := components[len(components)-2]
+		curPETM, err := NewS3RepositoryProtectedEntityTypeManager(curTypeName, session, bucket, prefix, logger)
+		if err != nil {
+
```
Comment (deepakkinni) [inline pkg/s3repository/repository_protected_entity_manager.go:68]: log the error, or even warning

**Review by deepakkinni — APPROVED** [review]
nits, looks good otherwise!

**Comment by reasonerjt** [comment]
Could you add more background in the commit message?  Like why we wanna make it handle multiple pe types, what are the types?

File: pkg/s3repository/repository_protected_entity_manager.go (hunk 2 above)
Comment (dsu-igeek) [reply pkg/s3repository/repository_protected_entity_manager.go:90]: It does need to be declared.  It's a go-ism.  When I used returnPETM,err := ... it makes a new variable masking the returnPETM that is declared outside the block.

File: pkg/s3repository/repository_protected_entity_manager.go (hunk 1 above)
Comment (dsu-igeek) [reply pkg/s3repository/repository_protected_entity_manager.go:53]: OK

File: pkg/s3repository/repository_protected_entity_manager.go (hunk 3 above)
Comment (lintongj) [reply pkg/s3repository/repository_protected_entity_manager.go:68]: +1

**Review by lintongj — APPROVED** [review]
Overall looks good. just nits

**Pull request merged**